tests/test_plot_manager.py

```diff
@@ -0,0 +1,185 @@
+            f.write(
+                convert_to_bytes(
+                    json.dumps(plot_manager.to_dict(), default=default_encode)))
+
+        with open(f"{ROOT_DIR}/single-model-ckpt/plot_manager.json", "rb") as f:
+            golden_plot_manager_dict = json.load(f)
+
+        golden_plot_manager = PlotManager.from_dict(
+            golden_plot_manager_dict, self._single_model_config,
+            self._single_model_result_manager)
+
+        golden = golden_plot_manager._simple_plots['add_sub'][
+            'throughput_v_latency']
+        plot = plot_manager._simple_plots['add_sub']['throughput_v_latency']
+        self._compare_plot_data(golden, plot)
+
+        golden = golden_plot_manager._simple_plots['add_sub'][
+            'gpu_mem_v_latency']
+        plot = plot_manager._simple_plots['add_sub']['gpu_mem_v_latency']
+        self._compare_plot_data(golden, plot)
+
+    def test_multi_model_summary_plots_against_golden(self):
+        """
+        Match the summary plots against the golden versions in
+        tests/common/multi-model-ckpt
+        """
+        plot_manager = PlotManager(
+            config=self._multi_model_config,
+            result_manager=self._multi_model_result_manager)
+
+        plot_manager.create_summary_plots()
+
+        # Uncomment these lines to create a new output dump to compare against
+        with open(f"{ROOT_DIR}/multi-model-ckpt/plot_manager.json", "wb") as f:
+            f.write(
+                convert_to_bytes(
+                    json.dumps(plot_manager.to_dict(), default=default_encode)))
+
+        with open(f"{ROOT_DIR}/multi-model-ckpt/plot_manager.json", "rb") as f:
+            golden_plot_manager_dict = json.load(f)
+
+        golden_plot_manager = PlotManager.from_dict(
+            golden_plot_manager_dict, self._multi_model_config,
+            self._multi_model_result_manager)
+
+        golden = golden_plot_manager._simple_plots[
+            'resnet50_libtorch,vgg19_libtorch']['throughput_v_latency']
+        plot = plot_manager._simple_plots['resnet50_libtorch,vgg19_libtorch'][
+            'throughput_v_latency']
+        self._compare_plot_data(golden, plot)
+
+        golden = golden_plot_manager._simple_plots[
+            'resnet50_libtorch,vgg19_libtorch']['gpu_mem_v_latency']
+        plot = plot_manager._simple_plots['resnet50_libtorch,vgg19_libtorch'][
+            'gpu_mem_v_latency']
+        self._compare_plot_data(golden, plot)
+
+    def _compare_plot_data(self, golden, plot):
+        self.assertEqual(golden._name, plot._name)
+        self.assertEqual(golden._title, plot._title)
+        self.assertEqual(golden._x_axis, plot._x_axis)
+        self.assertEqual(golden._y_axis, plot._y_axis)
+        self.assertEqual(golden._monotonic, plot._monotonic)
+        self.assertEqual(golden._data, plot._data)
+
+        # These are generated by pyplot and will vary from run to run
```

Comment:
These comments can go away? They aren't part of the structure anymore?

Reply:
Yes. I've removed this.